lib/preprocessors/glimmer.js

```diff
@@ -0,0 +1,117 @@
+'use strict';
+
+const { getTemplateLocals } = require('@glimmer/syntax');
+const {
+  preprocessEmbeddedTemplates,
+} = require('ember-template-imports/lib/preprocess-embedded-templates');
+const util = require('ember-template-imports/src/util');
+
+const TRANSFORM_CACHE = new Map();
+const TEXT_CACHE = new Map();
+
+function gjs(text, filename) {
+  const isGlimmerFile = filename.endsWith('.gjs') || filename.endsWith('.gts');
+
+  if (!isGlimmerFile) {
+    return [
+      {
+        filename,
+        text,
+      },
+    ];
+  }
+
+  if (text.includes(util.TEMPLATE_TAG_PLACEHOLDER)) {
+    return [
+      {
+        filename,
+        text,
+      },
+    ];
+  }
+
+  const preprocessed = preprocessEmbeddedTemplates(text, {
+    getTemplateLocals,
+    relativePath: filename,
+
+    templateTag: util.TEMPLATE_TAG_NAME,
+    templateTagReplacement: util.TEMPLATE_TAG_PLACEHOLDER,
+
+    includeSourceMaps: false,
+    includeTemplateTokens: true,
+  });
+
+  // Because the output looks like
+  // `, { strictMode: true, scope: () => ({on,noop}) })]
+  // we could accidentally have this line formatted with
+  // prettier, or trailing commas, or any variety of formatting rules.
+  //
+  // We need to disable all of eslint around this "scope" return
+  // value and allow-list the rules that check for undefined identifiers
+  const transformed = preprocessed.output;
+
+  TRANSFORM_CACHE.set(filename, transformed);
+  TEXT_CACHE.set(filename, text);
+
+  return [
+    {
+      filename,
+      text: transformed,
+    },
+  ];
+}
+
+function mapRange(messages, filename) {
```

Comment:
Can you add a high-level comment to this function explaining what it does?

Reply:
done